field/goldilocks/goldilocks.go

```diff
@@ -153,30 +158,53 @@ func RandArray(count int) []Element {
 	return ret
 }
 
-func Add(elems ...Element) Element {
-	res := g.NewElement(0)
-	for _, elem := range elems {
-		res.Add(&res, &elem)
-	}
-	return res
+func Add(a, b Element) Element {
+	a.Add(&a, &b)
+	return a
+}
+
+func AddThree(a, b, c Element) Element {
+	a.Add(&a, &b)
+	a.Add(&a, &c)
+	return a
+}
+
+func AddFour(a, b, c, d Element) Element {
+	a.Add(&a, &b)
+	c.Add(&c, &d)
+	a.Add(&a, &c)
+	return a
+}
+
+func AddFive(a, b, c, d, e Element) Element {
+	a.Add(&a, &b)
+	c.Add(&c, &d)
+	a.Add(&a, &c)
+	a.Add(&a, &e)
+	return a
 }
 
 func Sub(a, b *Element) Element {
-	res := g.NewElement(0)
+	res := G_ZERO
 	res.Sub(a, b)
 	return res
 }
```

Comment:
Do we need this kind of functions? Why not use `a.Sub(a, b)` directly? If we don't want to mutate `a` I think we can copy `a` where we call it, this is not related to this PR but a general question

Reply:
This resulted in many many variable copies in the higher levels (like extension or curve impls). We can do that, too, yes.